addons/binding/org.openhab.binding.networkcamera/src/main/java/org/openhab/binding/networkcamera/internal/NetworkCameraHandlerFactory.java

```diff
@@ -0,0 +1,111 @@
+import org.osgi.service.component.ComponentContext;
+import org.osgi.service.component.annotations.Component;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link NetworkCameraHandlerFactory} is responsible for creating things and thing
+ * handlers.
+ *
+ * @author Pauli Anttila - Initial contribution
+ */
+@Component(service = ThingHandlerFactory.class, immediate = true, configurationPid = "binding.networkcamera")
+public class NetworkCameraHandlerFactory extends BaseThingHandlerFactory {
+    private Logger logger = LoggerFactory.getLogger(NetworkCameraHandlerFactory.class);
+
+    private final static Set<ThingTypeUID> SUPPORTED_THING_TYPES_UIDS = Collections
+            .singleton(THING_TYPE_MOTIONDETECTION);
+
+    private final int DEFAULT_PORT = 2121;
+    private final int DEFAULT_IDLE_TIMEOUT = 60;
+
+    private FtpServer ftpServer = new FtpServer();
+
+    @Override
+    public boolean supportsThingType(ThingTypeUID thingTypeUID) {
+        return SUPPORTED_THING_TYPES_UIDS.contains(thingTypeUID);
+    }
+
+    @Override
+    protected ThingHandler createHandler(Thing thing) {
+
+        ThingTypeUID thingTypeUID = thing.getThingTypeUID();
+
+        if (thingTypeUID.equals(THING_TYPE_MOTIONDETECTION)) {
+            return new NetworkCameraHandler(thing, ftpServer);
+        }
+
+        return null;
+    }
+
+    @Override
+    protected synchronized void activate(ComponentContext componentContext) {
+        super.activate(componentContext);
+        modified(componentContext);
+    }
+
+    @Override
+    protected synchronized void deactivate(ComponentContext componentContext) {
+        ftpServer.stopServer();
+        super.deactivate(componentContext);
+    }
+
+    protected synchronized void modified(ComponentContext componentContext) {
+        ftpServer.stopServer();
+
+        Dictionary<String, Object> properties = componentContext.getProperties();
+
+        int port = DEFAULT_PORT;
+        int idleTimeout = DEFAULT_IDLE_TIMEOUT;
+
+        String strPort = (String) properties.get("port");
+        String strIdleTimeout = (String) componentContext.getProperties().get("idleTimeout");
+
+        if (StringUtils.isNotEmpty(strPort)) {
+            try {
+                port = Integer.valueOf(strPort);
+            } catch (NumberFormatException e) {
+                logger.warn("Invalid port number '{}', using default port {}", strPort, port);
+            }
+        }
+
+        if (StringUtils.isNotEmpty(strPort)) {
+            try {
+                idleTimeout = Integer.valueOf(strIdleTimeout);
+            } catch (NumberFormatException e) {
+                logger.warn("Invalid idle timeout '{}', using default timeout {}", strIdleTimeout, idleTimeout);
+            }
+        }
+
+        try {
+            ftpServer.startServer(port, idleTimeout);
+        } catch (FtpException e) {
+            logger.warn("FTP server starting failed, reason: {}", e.getMessage());
```

Comment:
As I mentioned before you should store all you created handlers in something like a `Set` and notify them that the ftpserver is gone if this happens. So they can set their things to offline.

Reply:
If e.g. FTP server port can't be bind, FTP server throws IOException. As it's not catch, bundle activation fails and things stays in uninitialised state. There is no mechanism that ftp server could notify that server is suddenly shutdown. I don't know when ftp server could throw FtpException in start, but if I remove FtpException catching, things should stay Offline state if server throws that exception.

Reply:
I think you should catch this Exception and not make the bundle initialization fail. This is an error case which can be treated by handling exception, so we should do it. Note that your `modified` method is called if you change the binding configuration, so you start the server not only on the initial bundle start but also because of a changed configuration.